### PR TITLE
applications: sdp: gpio: add Hard RealTime IRQ code

### PR DESCRIPTION
**Pull request by masz-nordic**
Depends on:
- https://github.com/nrfconnect/sdk-nrf/pull/17838
- https://github.com/nrfconnect/sdk-nrf/pull/17638

### CODE REVIEW

File: applications/sdp/gpio/src/main.c [hunk 1]
```diff
@@ -13,6 +15,16 @@
 #include <hal/nrf_vpr_csr_vio.h>
 #include <haly/nrfy_gpio.h>
 
+#define HRT_IRQ_PRIORITY      2
+#define HRT_VEVIF_GPIO_CLEAR  17
+#define HRT_VEVIF_GPIO_SET    18
+#define HRT_VEVIF_GPIO_TOGGLE 19
```
Comment (magp-nordic) [inline applications/sdp/gpio/src/main.c:21]: Maybe these names should include IRQ, for example, `HRT_IRQ_VEVIF_GPIO_CLEAR`? Right now, just by reading these lines, it is difficult to tell what those are for.

**Review by magp-nordic — COMMENTED** [review]
It should most probably be stated somewhere, which synchronization mechanism between the non-real-time and the hard real-time is used here, but I guess it can be done later with/in documentation.

File: applications/sdp/gpio/src/main.c (hunk 1 above)
Comment (masz-nordic) [reply applications/sdp/gpio/src/main.c:21]: My aim here was to show that VEVIF index is not identical to IRQ number.
Using `HRT_IRQ_...` could be confusing when there is `VEVIF_IRQN` macro.
Maybe `HRT_VEVIF_IDX_GPIO_CLEAR` would be better?

Comment (magp-nordic) [reply applications/sdp/gpio/src/main.c:21]: Ok, my bad, I thought those were IRQ numbers. But yes, I think adding `IDX` would make it more clear anyway

Comment (masz-nordic) [reply applications/sdp/gpio/src/main.c]: Done

**Review by carlescufi — APPROVED** [review]

**Pull request merged**